doc/classes/NetworkedMultiplayerCustom.xml

```diff
@@ -6,6 +6,70 @@
+		    get_tree().network_peer = custom_peer
+
+		    # Start connecting via real transport (ex. Steam peer-to-peer, PlayFab, etc).
+		    real_multiplayer_client = _connect_to_real_multiplayer()
+		    real_multiplayer_client.connect("connection_succeeded", self, "_on_real_multiplayer_connection_succeeded")
+		    real_multiplayer_client.connect("connection_failed", self, "_on_real_multiplayer_connection_failed")
+		    real_multiplayer_client.connect("peer_connected", self, "_on_real_multiplayer_peer_connected")
+		    real_multiplayer_client.connect("peer_disconnected", self, "_on_real_multiplayer_peer_disconnected")
+		    real_multiplayer_client.connect("recieved_message", self, "_on_real_multiplayer_received_message")
+
+		    # Intercept any RPC's sent locally.
+		    custom_peer.connect("packet_generated", self, "_on_custom_peer_packet_generated")
+
+		func _on_real_multiplayer_connection_succeeded():
+		    # Once connection is formed, assign a peer id somehow. If the real transport
+		    # doesn't have a compatible id, you'll need to generate your own ids, and map
+		    # them to the ids used by the real transport (the same mapping should be used
+		    # by all peers).
+		    #
+		    # Also, if you want Godot to treat this peer as the "host" or "server" (per
+		    # `get_tree().is_network_server()`) then use the peer id of 1.
+		    var my_id = real_multiplayer_client.get_my_id()
+		    custom_peer.initialize(my_id)
+		    custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_CONNECTED)
+
+		func _on_real_multiplayer_connection_failed():
+		    custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_DISCONNECTED)
+
+		func _on_real_multiplayer_peer_connected(peer_id):
+		    # Let Godot know a peer connected. Again, you may need to map between Godot ids
+		    # and the ids used by the real multiplayer client.
+		    custom_peer.emit_signal("peer_connected", peer_id)
+
+		func _on_real_multiplayer_peer_disconnected(peer_id):
+		    # Same notes as above.
+		    custom_peer.emit_signal("peer_disconnected", peer_id)
+
+		func _on_custom_peer_packet_generated(peer_id, buffer, transfer_mode):
+		    # Send any packets generated locally (ex RPCs) to the requested peer
+		    # using the real multiplayer client.
+		    if transfer_mode == NetworkedMultiplayerPeer.TRANSFER_MODE_RELIABLE:
+		        real_multiplayer_client.send_reliable(peer_id, {
+		            'from_peer_id': custom_peer.get_unique_id(),
+		            'data': buffer,
+		        })
+		    else:
+		        real_multiplayer_client.send_unreliable(peer_id, {
+		            'from_peer_id': custom_peer.get_unique_id(),
+		            'data': buffer,
+		        })
+
+		func _on_real_multiplayer_received_message(message):
+		    # Take any packets we recieved from other peers, and deliver them locally.
+		    custom_peer.deliver_packet(message['data'], message['from_peer_id'])
```

Comment:
I finally got Discord's Game SDK stuff figured out and have the packets being received by Godot. Now, however, I'm having issues with delivering the packets. I'm getting the `ERROR: Invalid packet received. Unabled to find requested cached node.` from the host when a client sends a packet using RPC. When the host sends the client a packet it's handled fine. If I ignore it and send another packet from the client, it seems to be processed ok. Not sure why the first packet is throwing that error. I've tested it using the ENet peer and that's still working so it is in my implementation of the custom network class that's causing issues. Any ideas or suggestions?

Reply:
Digging deeper here and I think it has something to do with how RPC calls are handled. So I call RPC from the client which gets passed to the server with the name of the function. That's it though, no node path is included. From the error, the docs, and sniffing about in the source code, it's acting like it should have cached the path of that node beforehand. But that was the first packet the server had received so how was it supposed to know where to execute that call from? Seems like something is missing but I can't figure out what. I'm wondering if `poll` is supposed to have any implementation in it? Or if I should call poll at some point?

Reply:
I suspect it's an ordering or race condition sort thing. Is it possible that the client is sending messages to the host before either the host or peer is fully ready? I had some issues like this while making my Nakama implementation, where the ID map wouldn't be fully setup before one of the various signals fired, which would lead to a peer trying to send a message, but it not arriving because it couldn't figure out the right ID. The solution was just making sure anything that would lead to emitting a signal happened after any internal bookkeeping.

If it's helpful, my Nakama implementation is here:

https://github.com/heroiclabs/nakama-godot/pull/116

> I'm wondering if poll is supposed to have any implementation in it? Or if I should call poll at some point?

There shouldn't be anything additional you need to do - if all the packets get delivered to the right peer, they'll have enough information to for the RPC to work.

Anyway, this conversation would probably be better on a separate support issue, or Discord/RocketChat or something! It'd be best to keep the comments here about getting this documentation fixed up. :-)

Reply:
Thanks for the advice. I'm making the call pretty late after the setup so I don't think it's a race condition in that way.
I'll create a separate issue for this. Just figured if it was something simple I was missing from the setup instructions we could get it ironed out here real quick.

Reply:
Here's the issue I created if you're interested in following it https://github.com/godotengine/godot/issues/64736

Reply:
Thanks! I've taken a quick look at your project, and commented over there